src/all/novelfull/src/eu/kanade/tachiyomi/extension/all/novelfull/NovelFullGlobal.kt

```diff
@@ -0,0 +1,241 @@
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/comic/bookclass.html?type=last_release&page_num=$page&language=$lang")
+    }
+
+    override fun latestUpdatesSelector(): String = popularMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector(): String? = popularMangaNextPageSelector()
+
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            return GET("$baseUrl/comic/search.html?keyword=$query&page_num=$page")
+        }
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreFilter -> {
+                    val filterId = filter.selected
+                    return GET("$baseUrl/comic/bookclass.html?type=category_novel&id=$filterId&page_num=$page&language=$lang")
+                }
+                else -> { }
+            }
+        }
+        TODO("Need to implement empty search somewhere")
+    }
+
+    override fun searchMangaSelector(): String = popularMangaSelector()
+
+    override fun searchMangaFromElement(element: Element): SManga = popularMangaFromElement(element)
+
+    override fun searchMangaNextPageSelector(): String? = popularMangaNextPageSelector()
+
+    override fun fetchSearchManga(
+        page: Int,
+        query: String,
+        filters: FilterList,
+    ): Observable<MangasPage> {
+        if (!supportsSearch && query.isNotEmpty()) {
+            return Observable.just(MangasPage(emptyList(), false))
+        }
+        return super.fetchSearchManga(page, query, filters)
+    }
+
+    // =============================== Filters ==============================
+
+    override fun getFilterList() = FilterList(
+        Note,
+        Filter.Separator(),
+        GenreFilter(getGenreFilter(lang), 0),
+    )
+
+    // =========================== Manga Details ============================
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        return SManga.create().apply {
+            title = document.selectFirst("h3.title")!!.text()
+            description = document.selectFirst(".desc-text")!!.text()
+                .replace("\\n", "\n")
+            thumbnail_url = document.selectFirst(".books > .book > img")!!.imgAttr()
+            document.selectFirst(Evaluator.Class("info"))!!.children().forEach {
+                when (it.children().first()!!.text()) {
+                    "Author:" -> it.parseAuthorsTo(this)
+                    "Genre:" -> it.parseGenresTo(this)
+                    "Status:" -> it.parseStatusTo(this)
+                }
+            }
+        }
+    }
+
+    private fun Element.parseAuthorsTo(manga: SManga) {
+        val authors = this.select("a").map { it.text() }
+        manga.author = authors.joinToString(", ")
```

Comment:
`, ` is the default and can be omitted
```suggestion
        manga.author = authors.joinToString()
```